src/fs.rs

```diff
@@ -110,3 +142,22 @@ fn read_dir_aux(path: &Path) -> std::io::Result<Vec<PathBuf>> {
     res.sort();
     Ok(res)
 }
+
+/// A temporary directory.
+#[derive(Debug)]
+pub struct TempDir {
+    path: PathBuf,
+}
+
+impl TempDir {
+    /// Returns the path of this temporary directory.
+    pub fn path(&self) -> &Path {
+        &self.path
+    }
+}
+
+impl Drop for TempDir {
+    fn drop(&mut self) {
+        rm_rf(&self.path).unwrap()
```

Comment:
```suggestion
        let _ = rm_rf(&self.path);
```

or maybe `if !std::thread::panicking() { .unwrap() } `

Reply:
Looks like [std is leery][1] of panicking in destructors, so let's do the former. But maybe it would be nice to circle back and add a `fn finish(self) -> Result<()>` method to all the RAII guards if one wants to catch errors and not drop them on the floor.

[1]: https://doc.rust-lang.org/stable/src/std/io/buffered/bufwriter.rs.html#413-421